app/org/maproulette/models/dal/ChallengeDAL.scala

```diff
@@ -1903,4 +1903,45 @@ class ChallengeDAL @Inject() (
       sqlWithParameters(query, parameters).as(this.withVirtualParentParser.*)
     }
   }
+
+  /**
+    * Retrieve tasks within a bounding box for a challenge, ordered by proximity
+    */
+  def getNearbyTasksWithinBoundingBox(
+      user: User,
+      challengeId: Long,
+      left: Double,
+      bottom: Double,
+      right: Double,
+      top: Double,
+      limit: Int = 5
+  )(
+      implicit c: Option[Connection] = None
+  ): List[Task] = {
+    this.permission.hasReadAccess(ChallengeType(), user)(challengeId)
+
+    // Get the center point of the bounding box for proximity calculations
+    val centerLat = (top + bottom) / 2
+    val centerLon = (left + right) / 2
+
+    val query = s"""SELECT tasks.${taskDAL.retrieveColumnsWithReview} FROM tasks
+      LEFT JOIN locked l ON l.item_id = tasks.id
+      LEFT OUTER JOIN task_review ON task_review.task_id = tasks.id
+      WHERE parent_id = $challengeId AND
+            tasks.location && ST_MakeEnvelope($left, $bottom, $right, $top, 4326) AND
+            (l.id IS NULL OR l.user_id = ${user.id}) AND
+            tasks.status IN (0, 3, 6) AND
+            NOT tasks.id IN (
+                SELECT task_id FROM status_actions
+                WHERE osm_user_id = ${user.osmProfile.id} AND created >= NOW() - '1 hour'::INTERVAL)
+      ORDER BY ST_Distance(
+        tasks.location, 
+        ST_SetSRID(ST_MakePoint($centerLon, $centerLat), 4326)
+      ), tasks.status, RANDOM()
```

Comment:
What's the reason for `ORDER BY ... RANDOM()`? Seems like tie-breaking on something deterministic (like `tasks.id`) would be preferable since it means that for a given query and DB state the results are deterministic (useful I imagine for unit tests).